### DISCUSSION TITLE
Data Resources as CKAN internal resource descriptors

### DISCUSSION THREAD

**Discussion by amercader** [frictionlessdata/ckanext-validation#72] [Ideas]
(prompted by [this comment](https://github.com/frictionlessdata/ckanext-validation/pull/69#issuecomment-1286862915))

> @amercader I have a more high-level question. Have you ever thought about adopting (Tabular) Data Resource Standard to be an internal part of CKAN for resource/file metadata description? We started a similar discussion with Dan Feder from DCAN and, currently, they are working on supporting something like this. In Frictionless Application, any resource is also described by Data Resource descriptor.

@roll do you mean changing the way resources are represented in the API and/or the internal model for resources? 
As much as I dislike some of the choices made in the resource representation it's part of an API that has been around forever, so changing things like `name` or `url` to make them fit the Data Resource standard would mean a new API version, which might be fine down the line, but a very strong case would have to be made for it:

```
      {
        "created": "2022-07-13T10:18:59.008556",
        "datastore_active": true,
        "description": "",
        "format": "CSV",
        "hash": "",
        "id": "85da2a1c-9532-4d79-a917-102a8c951053",
        "last_modified": "2022-07-13T10:18:58.975731",
        "metadata_modified": "2022-07-13T10:19:00.128392",
        "mimetype": "text/csv",
        "mimetype_inner": null,
        "name": "May 2022.csv",
        "package_id": "d96e47f2-db9e-4337-8641-075df7842700",
        "position": 0,
        "resource_type": null,
        "size": 1024324,
        "state": "active",
        "url": "https://demo.ckan.org/dataset/d96e47f2-db9e-4337-8641-075df7842700/resource/85da2a1c-9532-4d79-a917-102a8c951053/download/may-2022.csv",
        "url_type": "upload"
      }
```

If you meant how resources are handled internally, could you give an example? I'm not clear on what you mean

Is the DKAN discussion publicly available? I'd be curious to check it

**Comment by roll** [comment]
@amercader 
I think exact naming is not really important (it can be done in e.g. import/export like `ckanext-datapackager` does) -- it's more about native support for Table Schema/Dialect and maybe some additional properties

The discussion is really hard-to-trace as it was in Slack - https://frictionlessdata.slack.com/archives/C0369HZ2SLT/p1661437372374169

**Comment by roll** [comment]
> If you meant how resources are handled internally, could you give an example? I'm not clear on what you mean

E.g. Frictionless Application db has a schema (simplified just to share the idea):
- path
- resource (jsonb) -- Data Resource Standard
- report (jsonb)
